### ISSUE TITLE
[prefer-namespace-import] why namespace over named import?

### ISSUE THREAD

**Issue by gajus** [marcalexiei/eslint-plugin-zod#172]
### 🧩 Description

I would think that named import

```ts
import { z } from 'zod';
```

is better than

```
import * as z from 'zod';
```

https://github.com/marcalexiei/eslint-plugin-zod-x/blob/main/docs/rules/prefer-namespace-import.md

docs mention 'tree shaking', but I don't see how that's helping here. 

### ✒️ Additional context

I would suggest to add an option to prefer named import.

**Comment by marcalexiei** [comment]
I've always used namespace import because the first example in the documentation uses it (I started using `zod` from `v4`):

https://zod.dev/basics

<details><summary>Reference</summary>
<p>

<img width="844" height="388" alt="Image" src="https://github.com/user-attachments/assets/69a7b256-e78b-47d2-b51a-f735a7c2d3f7" />

</p>
</details>

## Additional findings

- https://github.com/samchungy/eslint-plugin-import-zod
- Comment from Colin Hacks highlighting that `next.js`, `webpack` and `esbuild` need namespace import to properly perform three shaking: https://github.com/colinhacks/zod/issues/4433#issuecomment-2921500831 
  (Probably I should add this to the rule documentation 😅)

---

> I would suggest to add an option to prefer named import.

Since the rule is named `prefer-namespace-import` I would rather implement a new rule named `consistent-import` with an option that allows the user to choose which import syntax use (`namespace` or `named`).
This will cause `prefer-namespace-import` to be deprecated and removed in a major version.

WDYT?
